src/views/validated-designs/guide/index.tsx

```diff
@@ -4,56 +4,140 @@
 
-import { HvdPage } from '../types'
+import s from './detail-view.module.css'
+
+import type { HvdPage } from '../types'
+import type { MDXRemoteSerializeResult } from 'next-mdx-remote'
+import type { OutlineLinkItem } from 'components/outline-nav/types'
 
 export interface ValidatedDesignsGuideProps {
 	title: string
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	mdxSource: any // @TODO: fix type
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	headers: any // @TODO: fix type
+	markdown: {
+		mdxSource: MDXRemoteSerializeResult
+		title: string
+		description: string
+	}
+	headers: OutlineLinkItem[]
 	currentPageIndex: number
-	categorySlug: string
 	basePath: string
 	pages: HvdPage[]
 }
 
 export default function ValidatedDesignGuideView({
 	title,
-	mdxSource,
+	markdown,
 	headers,
 	currentPageIndex,
 	basePath,
 	pages,
 }: ValidatedDesignsGuideProps) {
+	const docsMdxComponents = getDocsMdxComponents('hcp')
+	const backBtnTitle = 'HVD Home'
+	const backBtnUrl = basePath
+
+	function renderPreviousLink() {
+		if (currentPageIndex !== 0 && pages[currentPageIndex - 1]) {
+			const page = pages[currentPageIndex - 1]
+
+			return (
+				<DirectionalLinkBox
+					label="Previous"
+					name={page.title}
+					href={page.href}
+					direction={'previous'}
+					ariaLabel={`Go to previous page: ${page.title}`}
+				/>
+			)
+		}
+	}
+
+	function renderNextLink() {
+		if (currentPageIndex !== pages.length - 1 && pages[currentPageIndex + 1]) {
+			const page = pages[currentPageIndex + 1]
+
+			return (
+				<DirectionalLinkBox
+					label="Next"
+					name={page.title}
+					href={page.href}
+					direction={'next'}
+					ariaLabel={`Go to next page: ${page.title}`}
+				/>
+			)
+		}
+	}
+
 	return (
 		<SidebarSidecarLayout
-			AlternateSidebar={() => (
-				<>
-					<InlineLink href={`${basePath}`}>
-						Back to Validated Designs landing
-					</InlineLink>
-					{pages.map((page: HvdPage, index: number) => (
-						<li key={page.slug}>
-							<InlineLink href={page.href}>{page.title}</InlineLink>
-							{index === currentPageIndex && <span> {'<- current'}</span>}
-						</li>
-					))}
-				</>
-			)}
-			sidebarNavDataLevels={[]}
-			// @TODO: move to using OutlineNavWithActiveLink
-			sidecarSlot={<OutlineNav items={headers} />}
+			sidebarNavDataLevels={[
+				{
+					backToLinkProps: {
+						text: backBtnTitle,
+						href: backBtnUrl,
+					},
+					title,
+					visuallyHideTitle: true,
+					showFilterInput: false,
+					showResourcesList: false,
+					children: (
+						<>
+							<SidebarSectionBrandedHeading text={title} theme={'hcp'} />
+							<SidebarHorizontalRule />
+							<SidebarNavList>
+								{pages.map((page: HvdPage, index: number) => {
+									return (
+										<SidebarNavHighlightItem
+											key={page.slug}
+											text={page.title}
+											theme={'generic'}
+											href={page.href}
+											isActive={index === currentPageIndex}
+										/>
```

Comment:
There's a weird bug where the first two links in the menu don't have any text:
<img width="1378" alt="Screenshot 2024-01-03 at 18 53 50" src="https://github.com/hashicorp/dev-portal/assets/55773810/faaa3779-afb2-4ec4-b45b-7f900adba7a2">